### PR TITLE
[match][sigh] speed up profiles fetching by adding name filter to the API calls

### PR DESCRIPTION
**Pull request by PaulTaykalo**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Speeding up some queries on AppStoreConnectAPI

### Description

When running `match` and `sigh` commands, and there's a need for searching specific provisioning profiles, the current general pattern is to search for all profiles by calling `Spaceship::ConnectAPI::Profile.all` and filter later i.e.
```
all_profiles = Spaceship::ConnectAPI::Profile.all(includes: "devices")
```

In this PR we're adding additional filtering when querying `ConnectAPI` which greatly reduces request times.
```
all_profiles = Spaceship::ConnectAPI::Profile.all(filter: { name: name}, includes: "devices")
```

These queries are running multiple times during `sigh`/`match` tools so decreasing time even for a few seconds will get a great improvement in speed 

### Important notes
- While it seems that profile can simply be renamed, and have the same `uuid`, it's not actually true. When the profile is renamed, its `uuid` changes as well. 
- There's no visible way to require exact name matching when querying, so we're always adding later checks and not relying on names only. For example, if "Profile" is searched, "Profile1" and "Profile2" will be returned.

### Some numbers
Numbers greatly depend on the portal size, these are numbers for a real one (~200 profiles, 50 certs)
Querying all profiles (without specifying name) was taking from 60 to 150 seconds.
Adding name filtering decreased fetching time by almost 25x times 😮 
```
INFO [2023-01-29 01:46:11.10]: Time for fetching all profiles Without names: 125.897404
INFO [2023-01-29 01:46:12.57]: Time for fetching all profiles with names: 1.467258
INFO [2023-01-29 01:48:43.37]: Time for fetching all profiles Without names: 150.727023
INFO [2023-01-29 01:48:49.80]: Time for fetching all profiles: 6.429851
```

### Testing Steps

Basically, `sigh` and `match` should work exactly the same.

### CODE REVIEW

**Review by gelosi — APPROVED** [review]
Wow. Can’t wait to upgrade match 👏

**Comment by nekrich** [comment]
@joshdholtz @rogerluan can you please merge this PR. We are using this improvement for 6 months already and are really happy with it.

**Review by rogerluan — APPROVED** [review]
💯 Nice!

**Review by rogerluan — CHANGES_REQUESTED** [review]
@PaulTaykalo , sorry that this PR didn't get merged yet 😬 it ended up developing merge conflicts, would you mind fixing them by merging master into this branch?

Also, since we've merged a PR that potentially improved match performance by up to 100x, would you mind running your tests again to check whether this PR still improves performance, and by how much? 🙇 

Thank you so much!

**Comment by PaulTaykalo** [comment]
@rogerluan from what I see all these changes were added in https://github.com/fastlane/fastlane/pull/21694 this PR
So, I'll close current PR

**Comment by rogerluan** [comment]
Thanks for verifying this, @PaulTaykalo 🤗
